sdk/storage/Azure.Storage.Files.DataLake/src/Models/PathPermissions.cs

```diff
@@ -0,0 +1,238 @@
+        /// directory may only be renamed or deleted by the file's owner, the directory's owner, or the root user.
+        /// </summary>
+        public bool StickyBit { get; set; }
+
+        /// <summary>
+        /// Whether or not there is more permissions information in the ACLs. The permissions string only returns
+        /// information on the owner, owning group, and other, but the ACLs may contain more permissions for specific users
+        /// or groups.
+        /// </summary>
+        public bool ExtendedInfoInAcl { get; set; }
+
+        /// <summary>
+        /// Internal empty constructor.
+        /// </summary>
+        public PathPermissions() { }
+
+        /// <summary>
+        /// Public constructor.
+        /// </summary>
+        /// <param name="owner">The path owner's permissions.</param>
+        /// <param name="group">The path group's permissions.</param>
+        /// <param name="other">Permissions for other users.</param>
+        /// <param name="stickyBit">If sticky bit is enabled</param>
+        /// <param name="extendedInfoInAcl">If there is extended info in the ACL</param>
+        public PathPermissions(
+            RolePermissions owner,
+            RolePermissions group,
+            RolePermissions other,
+            bool stickyBit,
+            bool extendedInfoInAcl)
+        {
+            Owner = owner;
+            Group = group;
+            Other = other;
+            StickyBit = stickyBit;
+            ExtendedInfoInAcl = extendedInfoInAcl;
+        }
+
+        /// <summary>
+        /// Parses a string in octal format to PathPermissions.
+        /// </summary>
+        /// <param name="octal">Octal string to parse.</param>
+        /// <returns><see cref="PathPermissions"/>.</returns>
+        public static PathPermissions ParseOctal(string octal)
+        {
+            if (octal == null)
+            {
+                throw Errors.ArgumentNull(nameof(octal));
+            }
+
+            if (octal.Length != 4)
+            {
+                throw Errors.InvalidArgument(nameof(octal));
+            }
+
+            var pathPermissions = new PathPermissions();
+
+            if (octal[0] == '0')
+            {
+                pathPermissions.StickyBit = false;
+            }
+            else
+            {
+                pathPermissions.StickyBit = true;
+            }
+
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[1]);
+            pathPermissions.Group = RolePermissionsExtensions.ParseOctal(octal[2]);
+            pathPermissions.Owner = RolePermissionsExtensions.ParseOctal(octal[3]);
+
+            return pathPermissions;
+        }
+
+        /// <summary>
+        /// Parses a symbolic string to PathPermissions.
+        /// </summary>
+        /// <param name="str">String to parse</param>
+        /// <returns><see cref="PathPermissions"/></returns>
+        public static PathPermissions ParseSymbolic(string str)
```

Comment:
Does this need to be public?

Reply:
I think it’s convenient for more sophisticated used to be able to call PathPermissions.ParseSymbolic(“rw—wx—x”) than use the constructor.